caffe2/operators/batch_gather_ops.h

```diff
@@ -22,15 +22,14 @@ class BatchGatherOp final : public Operator<Context> {
   bool DoRunWithType() {
     auto& data = Input(DATA);
     auto& indices = Input(INDICES);
-    auto* output = Output(0);
 
     CAFFE_ENFORCE_GE(data.dim(), 2, "DATA should be at least 2-D");
 
     vector<int64_t> shape;
     shape.push_back(data.size(0));
     shape.insert(shape.end(), indices.sizes().begin(), indices.sizes().end());
     shape.insert(shape.end(), data.sizes().begin() + 2, data.sizes().end());
-    output->Resize(shape);
+    auto* output = Output(0, shape, at::dtype<float>());
```

Comment:
The code below doesn't look like this is always float. It only uses float if data.IsType<float>(). It uses data.dtype(), maybe you should use that too?